Roles needed to run rpCheckup

What role(s) does the user require in order to run all the checks? I'm getting the following error:

```
sudo go run main.go
INFO[0000] Running postgres                             
INFO[0000] Waiting for postgres to be healthy           
INFO[0032] Postgres is healthy                          
panic: no EC2 IMDS role found, operation error ec2imds: GetMetadata, http response error StatusCode: 404, request to EC2 IMDS failed

goroutine 1 [running]:
main.main()
	/home/ec2-user/rpCheckup/main.go:214 +0xef3
exit status 2
```
From experience typically when running such tools the SecurityAudit and/or Read-Only is required.

So I have my AWS creds in the .aws/credentials file, including any info in the config file. I. can run the prowler tool, with thte same credentials, without any issues. The user I'm using has substantial privileges, so that shouldn't be an issue.  
I use `sudo` since it's a rough install and it needs docker for postgress. Docker by default needs root, correct? 

I'm still getting the following error. I'm not a big goLang user, so not sure if Go isn't seeing my aws creds or what. 
```
 sudo go run main.go
INFO[0000] Running postgres                             
INFO[0001] Waiting for postgres to be healthy           
INFO[0033] Postgres is healthy                          
panic: no EC2 IMDS role found, operation error ec2imds: GetMetadata, http response error StatusCode: 404, request to EC2 IMDS failed

goroutine 1 [running]:
main.main()
	/home/ec2-user/rpCheckup/main.go:214 +0xef3
exit status 2
```

Received this error earlier:
```
go run main.go
INFO[0000] Running postgres                             
panic: Failed to pull image: Got permission denied while trying to connect to the Docker daemon socket at unix:///var/run/docker.sock: Post "http://%2Fvar%2Frun%2Fdocker.sock/v1.24/images/create?fromImage=postgres&tag=13-alpine": dial unix /var/run/docker.sock: connect: permission denied

goroutine 1 [running]:
main.main()
	/home/ec2-user/rpCheckup/main.go:209 +0xf05
exit status 2
```

--- a/main.go
+++ b/main.go
@@ -25,14 +25,26 @@ import (
 	"github.com/goldfiglabs/rpcheckup/pkg/report"
 )
 
+type awsAuthError struct {
+	Err error
+}
+
+func (e *awsAuthError) Error() string {
+	return "Failed to find AWS Credentials"
+}
+
+func (e *awsAuthError) Unwrap() error {
+	return e.Err
+}
+
 func loadAwsCredentials(ctx context.Context) ([]string, error) {
 	cfg, err := config.LoadDefaultConfig(ctx)
 	if err != nil {
-		return nil, err
+		return nil, &awsAuthError{err}
 	}
 	creds, err := cfg.Credentials.Retrieve(ctx)
 	if err != nil {
-		return nil, err
+		return nil, &awsAuthError{err}
 	}
 	env := []string{
 		fmt.Sprintf("AWS_ACCESS_KEY_ID=%v", creds.AccessKeyID),
@@ -209,10 +221,24 @@ func main() {
 	if err != nil {
 		panic(err)
 	}
+	shutdownPostgres := func() {
+		if !leavePostgresUp {
+			err = postgresService.ShutDown()
+			if err != nil {
+				panic(err)
+			}
+		}
+	}
 	if !skipIntrospector {
 		awsCreds, err := loadAwsCredentials(ds.Ctx)
 		if err != nil {
-			panic(err)
+			var authErr *awsAuthError
+			if errors.As(err, &authErr) {
+				shutdownPostgres()
+				log.Fatal("Failed to find AWS Credentials. Please ensure that your enviroment is correctly configued as described here: https://docs.aws.amazon.com/cli/latest/userguide/cli-chap-configure.html")
+			} else {
+				panic(err)
+			}
 		}
 		i, err := introspector.New(ds, postgresService, introspector.Options{
 			LogDockerOutput: logIntrospector,
@@ -256,10 +282,5 @@ func main() {
 		panic(err)
 	}
 	log.Infof("Reports written to directory %v", outputDir)
-	if !leavePostgresUp {
-		err = postgresService.ShutDown()
-		if err != nil {
-			panic(err)
-		}
-	}
+	shutdownPostgres()
 }

